src/recaptcha-verification/recaptcha-verification.service.ts

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { ApiConfigService } from '../api-config/api-config.service';
+
+@Injectable()
+export class RecaptchaVerificationService {
+  constructor(private readonly configService: ApiConfigService) {}
+
+  async verify(recaptcha?: string | undefined): Promise<boolean> {
+    if (recaptcha === undefined) {
+      throw new HttpException(
+        "Missing 'recaptcha' field",
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const recaptchaSecret = this.configService.get<string>(
+      'RECAPTCHA_SECRET_KEY',
+    );
+
+    const recaptchaVerificationUrl = `https://www.google.com/recaptcha/api/siteverify?secret=${recaptchaSecret}&response=${recaptcha}`;
+
+    const isVerified = await fetch(recaptchaVerificationUrl)
+      .then((response) => response.json())
```

Comment:
If you're doing an await, you don't need to put any of this code into the then(). you can just move it out into the main code area.

const response = await fetch() 
const body = response.json() as {{
          success: boolean;
          challenge_ts: Date;
          hostname: string;
          'error-codes'?: string[];
        }